abis_mapping/templates/survey_site_data_v2/mapping.py

```diff
@@ -142,10 +146,17 @@ def extract_geometry_defaults(
         # Context manager for row streaming
         with resource.open() as r:
             # Create empty dictionary to hold mapping values
-            result = {}
+            result: dict[str, str] = {}
             for row in r.row_stream:
                 # Extract values
-                site_id: str = row["siteID"]
+                site_id = _string_or_none.validate_python(row["siteID"])
+
+                # Even though siteID is a mandatory field, it can be missing here
+                # because this method is called on an invalid template for cross-validation,
+                # Skip this row if no siteID provided.
```

Comment:
How about 
```
# Check for siteID, even though siteID is a mandatory field, it can be missing here
# because this method is called for cross-validation, regardless of if this template is valid.
```

Reply:
all good will approve.